Libplanet/Blockchain/BlockChain.cs

```diff
@@ -1231,9 +1231,25 @@ internal BlockLocator GetBlockLocator(int threshold = 10)
             }
         }
 
-        internal ImmutableList<Transaction<T>> ListStagedTransactions()
+        /// <summary>
+        /// Lists the all staged transactions, with properly ordered nonces.
+        /// </summary>
+        /// <param name="txPriority">An optional comparer for give certain transactions to
+        /// priority to belong to the block.  No certain priority by default.</param>
+        /// <returns>A list of staged transactions.  This guarantees that for transactions signed
+        /// by the same address, those with greater nonce never comes before those with
+        /// lesser nonce.</returns>
+        internal ImmutableList<Transaction<T>> ListStagedTransactions(
+            IComparer<Transaction<T>> txPriority = null
```

Comment:
Should `txPriority` be propagated all the way down to `ListStagedTransactions`? I think it is more natural to handle this as transaction selection strategy in `GatherTransactionsToMine()`.

Reply:
I just want to prevent `txPriority` accidentally reordering tx nonces per signer, which is `ListStagedTransactions()`'s key guarantee.  Applying `txPriority` before `ListStagedTransactions()` ordering txs was my immediate idea to achieve it, but I'm going to consider it if you suggest another idea.

Reply:
Fair enough. I still have some qualms about "in place" nonce sorting's fairness in the first place, but that's a separate issue and I can't come up with a better solution at the moment. 🙃